cmd/src/debug_comp.go

```diff
@@ -0,0 +1,255 @@
+		flagSet: flagSet,
+		handler: handler,
+		usageFunc: func() {
+			fmt.Println(usage)
+		},
+	})
+}
+
+/*
+Docker functions
+*/
+
+func archiveCompose(ctx context.Context, zw *zip.Writer, verbose, configs bool, baseDir string) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	containers, err := getContainers(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get docker containers: %w", err)
+	}
+
+	if verbose {
+		log.Printf("getting docker data for %d containers...\n", len(containers))
+	}
+
+	// setup channel for slice of archive function outputs
+	ch := make(chan *archiveFile)
+	g, ctx := errgroup.WithContext(ctx)
+	semaphore := semaphore.NewWeighted(8)
+
+	// start goroutine to run docker ps -o wide
+	g.Go(func() error {
+		if err := semaphore.Acquire(ctx, 1); err != nil {
+			return err
+		}
+		defer semaphore.Release(1)
+		ch <- getPs(ctx, baseDir)
+		return nil
+	})
+
+	// start goroutine to run docker container stats --no-stream
+	g.Go(func() error {
+		if err := semaphore.Acquire(ctx, 1); err != nil {
+			return err
+		}
+		defer semaphore.Release(1)
+		ch <- getStats(ctx, baseDir)
+		return nil
+	})
+
+	// start goroutine to run docker container logs <container>
+	for _, container := range containers {
+		c := container
+		g.Go(func() error {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return err
+			}
+			defer semaphore.Release(1)
+			ch <- getContainerLog(ctx, c, baseDir)
+			return nil
+		})
+	}
+
+	// start goroutine to run docker container inspect <container>
+	for _, container := range containers {
+		c := container
+		g.Go(func() error {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return err
+			}
+			defer semaphore.Release(1)
+			ch <- getInspect(ctx, c, baseDir)
+			return nil
+		})
+	}
+
+	// start goroutine to get configs
+	if configs {
+		g.Go(func() error {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return err
+			}
+			defer semaphore.Release(1)
+			ch <- getExternalServicesConfig(ctx, baseDir)
+			return nil
+		})
+
+		g.Go(func() error {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return err
+			}
+			defer semaphore.Release(1)
+			ch <- getSiteConfig(ctx, baseDir)
+			return nil
+		})
+	}
+
+	// close channel when wait group goroutines have completed
+	go func() {
+		g.Wait()
+		close(ch)
+	}()
+
+	// Read binaries from channel and write to archive on host machine
+	if err := writeChannelContentsToZip(zw, ch, verbose); err != nil {
+		return fmt.Errorf("failed to write archives from channel: %w", err)
+	}
+
+	return nil
+}
+
+func getContainers(ctx context.Context) ([]string, error) {
+	c, err := exec.CommandContext(ctx, "docker", "container", "ls", "--format", "{{.Names}} {{.Networks}}").Output()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get container names with error: %w", err)
+	}
+	s := string(c)
+	preprocessed := strings.Split(strings.TrimSpace(s), "\n")
+	containers := []string{}
```

Comment:
```suggestion
	containers := make(string, 0, len(preprocessed))
```

You actually know the size here so you can make it with enough space up-front. This will reduce the number of re-allocations as the backing slice needs to grow.